src/core/net/tcp6.cpp

```diff
@@ -235,7 +235,11 @@ Error Tcp::Endpoint::ReceiveByReference(const otLinkedBuffer *&aBuffer)
 
 Error Tcp::Endpoint::ReceiveContiguify(void)
 {
-    return kErrorNotImplemented;
+    struct tcpcb &tp = GetTcb();
+
+    cbuf_contiguify(&tp.recvbuf, tp.reassbmp);
+
+    return kErrorNone;
```

Comment:
Maybe in a new/future PR. How about adding some test-cases to cover the `Contiguify` behavior (or maybe unit test for specific parts of it)?.